IntegrationTests/Services/AWSS3IntegrationTests/S3SigV4ATests.swift

```diff
@@ -54,84 +57,28 @@ class S3SigV4ATests: S3XCTestCase {
         stsClient = try STSClient(region: region)
         accountId = try await stsClient.getCallerIdentity(input: GetCallerIdentityInput()).account
 
-        // Construct MRAP config
-        let publicAccessBlock = S3ControlClientTypes.PublicAccessBlockConfiguration(
-            blockPublicAcls: false,
-            blockPublicPolicy: false,
-            ignorePublicAcls: false,
-            restrictPublicBuckets: false
-        )
-        let regions = [S3ControlClientTypes.Region(bucket: bucketName, bucketAccountId: accountId)]
-        let mrapConfig = S3ControlClientTypes.CreateMultiRegionAccessPointInput(
-            name: mrapName,
-            publicAccessBlock: publicAccessBlock,
-            regions: regions
-        )
-
-        // Create S3 Multi-Region Access Point (MRAP)
-        s3ControlClient = try S3ControlClient(region: region)
-        let createMRAPInput = CreateMultiRegionAccessPointInput(
-            accountId: accountId,
-            clientToken: UUID().uuidString.split(separator: "-").first!.lowercased(),
-            details: mrapConfig
-        )
-        _ = try await s3ControlClient.createMultiRegionAccessPoint(input: createMRAPInput).requestTokenARN
-
-        // Wait until MRAP creation finishes
-        var status: S3ControlClientTypes.MultiRegionAccessPointStatus? = .creating
-        repeat {
-            let seconds = 20.0
-            try await Task.sleep(nanoseconds: UInt64(seconds * Double(NSEC_PER_SEC)))
-
-            status = try await s3ControlClient.getMultiRegionAccessPoint(input: GetMultiRegionAccessPointInput(
-                accountId: accountId,
-                name: mrapName
-            )).accessPoint?.status
-        } while status == .creating
-
-        // Fetch MRAP alias then format & save MRAP ARN
-        // This MRAP ARN is used in place of bucket name in subsequent calls
-        mrapAlias = try await s3ControlClient.getMultiRegionAccessPoint(input: GetMultiRegionAccessPointInput(
-            accountId: accountId,
-            name: mrapName
-        )).accessPoint?.alias
-        mrapArn = String(format: mrapArnFormat, accountId, mrapAlias)
+        // Save mrapArn without creating new MRAP if it already exists.
+        if let preexistingMRAP = try await checkIfMRAPExists(), let alias = preexistingMRAP.alias {
+            mrapArn = String(format: mrapArnFormat, accountId, alias)
+        } else {
+            // Otherwise, create a new MRAP.
+            // Create a bucket to be linked to MRAP at construction.
+            try await super.setUp()
+            // Construct MRAP and save its ARN for use in tests.
+            _ = try await constructMRAP()
+        }
     }
 
     override func tearDown() async throws {
-        // MRAP must be deleted before deleting the linked bucket
-        try await deleteMRAP()
-        // Empty & delete the bucket
-        try await super.tearDown()
```

Comment:
This test class inherits `S3XCTestCase`, which does have a tear-down... we do call `super.setUp()` at least in some cases, which creates a test bucket... do we need to call `super.tearDown()` as well?

Reply:
The `super.tearDown()` deletes the bucket that was created in `super.setUp()`, but since we're keeping the MRAP, we can't delete the bucket. To delete the bucket, we have to delete MRAP first. Otherwise it would result in an error. That's why the it's a no-op now instead of deleting MRAP then the bucket.